libvast_test/vast/test/test.hpp

```diff
@@ -39,21 +50,29 @@
 // -- macros for checking results ----------------------------------------------
 
 #define REQUIRE CAF_REQUIRE
-#define REQUIRE_EQUAL CAF_REQUIRE_EQUAL
+#define REQUIRE_EQUAL(x, y) REQUIRE(x == y)
 #define REQUIRE_NOT_EQUAL CAF_REQUIRE_NOT_EQUAL
 #define REQUIRE_LESS CAF_REQUIRE_LESS
 #define REQUIRE_LESS_EQUAL CAF_REQUIRE_LESS_EQUAL
 #define REQUIRE_GREATER CAF_REQUIRE_GREATER
 #define REQUIRE_GREATER_EQUAL CAF_REQUIRE_GREATER_EQUAL
 #define CHECK CAF_CHECK
-#define CHECK_EQUAL CAF_CHECK_EQUAL
-#define CHECK_NOT_EQUAL CAF_CHECK_NOT_EQUAL
-#define CHECK_LESS CAF_CHECK_LESS
-#define CHECK_LESS_EQUAL CAF_CHECK_LESS_OR_EQUAL
-#define CHECK_GREATER CAF_CHECK_GREATER
-#define CHECK_GREATER_EQUAL CAF_CHECK_GREATER_OR_EQUAL
+#define CHECK_EQUAL(x, y)                                                      \
+  CAF_CHECK_FUNC(vast::test::detail::equality_compare, x, y)
```

Comment:
This variant looked the best to me, but I wanted to get a design review first before creating comparison structs for all the other operators and do the same for `REQUIRE_*`

Reply:
I think this reflects the expected behavior best. 

Can you add parentheses around the macro arguments?

```
#define REQUIRE_EQUAL(x, y) REQUIRE((x) == (y))
```

Reply:
Will this preserve the introspection behavior in case of a mismatch? If yes, then this is the way to go. The `REQUIRE_*` stuff will also need the same treatment.

Reply:
@tobim Yes, the output will be the same as before